packages/utils/auth.ts

```diff
@@ -0,0 +1,54 @@
+import { AdminNavRoutes, Routes } from './routes';
+
+export interface AuthConfig {
+  needAuth?: { permissions?: Permissions };
+}
+
+export type ComponentWithAuth<PropTypes = Record<string, never>> = React.FC<PropTypes> & AuthConfig;
+
+export enum SessionStatus {
+  LOADING = 'loading',
+  AUTHENTICATED = 'authenticated',
+  UNAUTHENTICATED = 'unauthenticated',
+}
+
+export enum AuthError {
+  CALLBACK = 'Callback',
+  SESSION_REQUIRED = 'SessionRequired',
+}
+
+export enum PermissionCategory {
+  USER = 'USER',
+  ROLE = 'ROLE',
+  EVENT = 'EVENT',
+}
+
+export enum PermissionPrivilege {
+  READ,
+  WRITE,
+}
+
+export type Permissions = {
+  [key in PermissionCategory]?: PermissionPrivilege;
+};
+
+export const ROUTES_TO_PERMISSIONS: Record<Routes | AdminNavRoutes, Permissions> = {
```

Comment:
Routes | AdminNavRoutes is confusing since AdminNavRoutes are routes ... maybe define a type who include all available routes